src/ansys/motorcad/core/geometry.py

```diff
@@ -415,6 +576,63 @@ def reverse(self):
 
         self.radius *= -1
 
+    def coordinate_on_entity(self, coordinate):
+        """Get if a coordinate exists on this Arc.
+
+        Parameters
+        ----------
+        coordinate : Coordinate
+            Check if this coordinate is on the Arc
+        Returns
+        -------
+        bool
+        """
+        v_from_centre = coordinate - self.centre
+        radius, angle_to_check = v_from_centre.get_polar_coords_deg()
+
+        theta1 = angle_to_check - self.start_angle()
+        theta2 = self.end_angle() - self.start_angle()
+
+        if self.radius > 0:
+            within_angle = theta1 <= theta2
+        else:
+            within_angle = theta2 <= theta1
+
+        return within_angle and (abs(radius) == abs(self.radius))
+
+    def start_angle(self):
+        """Get angle of start point from centre point coordinates.
+
+        Returns
+        -------
+        real
+        """
+        _, ang = (self.start - self.centre).get_polar_coords_deg()
+        return ang
+
+    def end_angle(self):
+        """Get angle of end point from centre point coordinates.
+
+        Returns
+        -------
+        real
+        """
+        _, ang = (self.end - self.centre).get_polar_coords_deg()
+        return ang
+
+    def total_angle(self):
+        """Get total angle that arc travels through. Assumes angle less than or equal to 180deg.
```

Comment:
I think this needs to be extended to arcs over 180 deg

Reply:
In hindsight there's a much easier way to test this. Updated it